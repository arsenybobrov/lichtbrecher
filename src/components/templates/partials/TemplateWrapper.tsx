import React from 'react';
import PrismicToolbar from '../../../thirdPartyScripts/PrismicToolbar';
import StylesWrapper from '../../../styles/StylesWrapper';
import ContextsWrapper from '../../../contexts/ContextsWrapper';
import { Data } from '../../../../prismic/types';
import HeadWrapper from '../../../head/HeadWrapper';

interface TemplateWrapperProps {
  pageId: string;
  data: Data;
  sharedData?: Data;
  serverReqUrl: string;
}

const TemplateWrapper: React.FC<TemplateWrapperProps> = ({
  pageId, data, sharedData, serverReqUrl, children,
}) => (
  <ContextsWrapper sharedData={sharedData}>
    <StylesWrapper>
      <HeadWrapper data={data} serverReqUrl={serverReqUrl}>
        <PrismicToolbar pageId={pageId} />
        {children}
      </HeadWrapper>
    </StylesWrapper>
  </ContextsWrapper>
);

export default TemplateWrapper;
