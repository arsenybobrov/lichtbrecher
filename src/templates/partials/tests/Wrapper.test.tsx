import React from 'react';
import renderer from 'react-test-renderer';
import TemplateWrapper from '../Wrapper';

const Data = {
  globalConfig: {
    id: 'XgeIahIAACEARC-h',
    uid: null,
    type: 'global_config',
    href: 'https://lichtbrecher.cdn.prismic.io/api/v2/documents/search?ref=XhnmmxUAACIAIDxH&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22XgeIahIAACEARC-h%22%29+%5D%5D',
    tags: [],
    first_publication_date: '2019-12-28T16:53:02+0000',
    last_publication_date: '2019-12-28T16:53:10+0000',
    slugs: [
      'globale-einstellungen',
    ],
    linked_documents: [],
    lang: 'de-de',
    alternate_languages: [
      {
        id: 'XgeLhBIAACEARD1W',
        type: 'global_config',
        lang: 'en-gb',
      },
    ],
    data: {
      display_name: [
        {
          type: 'heading1',
          text: 'Globale Einstellungen',
          spans: [],
        },
      ],
    },
  },
  data: {
    id: 'XgeHwRIAACMARCyx',
    uid: null,
    type: 'homepage',
    href: 'https://lichtbrecher.cdn.prismic.io/api/v2/documents/search?ref=XhnmmxUAACIAIDxH&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22XgeHwRIAACMARCyx%22%29+%5D%5D',
    tags: [
      'homepage',
    ],
    first_publication_date: '2019-12-28T16:50:15+0000',
    last_publication_date: '2020-01-11T15:15:39+0000',
    slugs: [
      'homepage',
    ],
    linked_documents: [],
    lang: 'de-de',
    alternate_languages: [
      {
        id: 'XgeLbBIAACQARDzp',
        uid: 'homepage',
        type: 'homepage',
        lang: 'en-gb',
      },
    ],
    data: {
      display_name: [
        {
          type: 'heading1',
          text: 'Homepage',
          spans: [],
        },
      ],
      headline: [
        {
          type: 'heading1',
          text: 'Ich bin die Startseite',
          spans: [],
        },
      ],
      text: [
        {
          type: 'paragraph',
          text: 'Auch gibt es niemanden, external der den Schmerz an sich liebt, sucht oder wünscht, nur, weil er Schmerz ist, es sei denn, es kommt zu zufälligen Umständen, in denen Mühen und Schmerz ihm große Freude bereiten können.',
          spans: [
            {
              start: 24,
              end: 32,
              type: 'hyperlink',
              data: {
                link_type: 'Web',
                url: 'https://example.com',
                target: '_blank',
              },
            },
            {
              start: 166,
              end: 183,
              type: 'hyperlink',
              data: {
                id: 'XgeIMBIAACMARC6d',
                type: 'page',
                tags: [
                  'one',
                ],
                slug: 'page-one',
                lang: 'de-de',
                uid: 'page-one',
                link_type: 'Document',
                isBroken: false,
              },
            },
          ],
        },
      ],
      linktext: 'PAGE TWO',
      url: {
        link_type: 'Web',
        url: 'https://example.com',
        target: '_blank',
      },
    },
  },
};

const Data404 = {
  globalConfig: {
    id: 'XgeIahIAACEARC-h',
    uid: null,
    type: 'global_config',
    href: 'https://lichtbrecher.cdn.prismic.io/api/v2/documents/search?ref=XhnmmxUAACIAIDxH&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22XgeIahIAACEARC-h%22%29+%5D%5D',
    tags: [],
    first_publication_date: '2019-12-28T16:53:02+0000',
    last_publication_date: '2019-12-28T16:53:10+0000',
    slugs: [
      'globale-einstellungen',
    ],
    linked_documents: [],
    lang: 'de-de',
    alternate_languages: [
      {
        id: 'XgeLhBIAACEARD1W',
        type: 'global_config',
        lang: 'en-gb',
      },
    ],
    data: {
      display_name: [
        {
          type: 'heading1',
          text: 'Globale Einstellungen',
          spans: [],
        },
      ],
    },
  },
};

const Data400 = {
  error400: true,
};

describe('templateWrapper snapshot', () => {
  it.each`
    name                     |  mock
    ${'default'}             |  ${Data}
    ${'Api is down 400'}     |  ${Data400}
    ${'Page not found 404'}  |  ${Data404}
  `('$name', ({ mock }) => {
  jest.spyOn(window, 'addEventListener').mockImplementation();
  const tree = renderer
    .create(
      <TemplateWrapper {...mock}>
        <h1>Hello world!</h1>
      </TemplateWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
});
