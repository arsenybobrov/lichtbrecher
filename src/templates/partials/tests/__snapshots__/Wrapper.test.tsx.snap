// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templateWrapper snapshot Api is down 400 1`] = `
Array [
  <h1>
    Error 400
  </h1>,
  <p>
    API is down
  </p>,
]
`;

exports[`templateWrapper snapshot Page not found 404 1`] = `
Array [
  <h1>
    404 TEMPLATE
  </h1>,
  <br />,
  <pre>
    {
  "id": "XgeIahIAACEARC-h",
  "uid": null,
  "type": "global_config",
  "href": "https://lichtbrecher.cdn.prismic.io/api/v2/documents/search?ref=XhnmmxUAACIAIDxH&q=%5B%5B%3Ad+%3D+at%28document.id%2C+%22XgeIahIAACEARC-h%22%29+%5D%5D",
  "tags": [],
  "first_publication_date": "2019-12-28T16:53:02+0000",
  "last_publication_date": "2019-12-28T16:53:10+0000",
  "slugs": [
    "globale-einstellungen"
  ],
  "linked_documents": [],
  "lang": "de-de",
  "alternate_languages": [
    {
      "id": "XgeLhBIAACEARD1W",
      "type": "global_config",
      "lang": "en-gb"
    }
  ],
  "data": {
    "display_name": [
      {
        "type": "heading1",
        "text": "Globale Einstellungen",
        "spans": []
      }
    ]
  }
}
  </pre>,
]
`;

exports[`templateWrapper snapshot default 1`] = `
<h1>
  Hello world!
</h1>
`;
